internal/common/utils.go

```diff
@@ -243,3 +239,50 @@ func GenerateDeleteOptions() *client.DeleteOptions {
 	}
 	return &delOpt
 }
+
+// RetryRoundTripper include anything here that's useful during middleware processing
+type RetryRoundTripper struct {
+	transport http.RoundTripper
+	backoff   wait.Backoff
+	log       logr.Logger
+}
+
+// RetryMiddleware pass this into your client to make it retriable with configured backoff
+func RetryMiddleware(logger logr.Logger) func(rt http.RoundTripper) http.RoundTripper {
+	return func(rt http.RoundTripper) http.RoundTripper {
+		return &RetryRoundTripper{
+			transport: rt,
+			backoff:   getBackoff(),
+			log:       logger.WithName("retry-middleware"),
+		}
+	}
+}
+
+// getBackoff configured for API outages where we require additional time to recover before being served.
+// this exponential with max wait time about ~3mins
+func getBackoff() wait.Backoff {
+	return wait.Backoff{
+		Steps:    45,
+		Duration: 10 * time.Millisecond,
+		Factor:   1.2,
+		Jitter:   0.1,
+	}
+}
+
+// RoundTrip implements RoundTripper to do retries.
+// This helpful especially when running in SNO,
+// where API server maybe down and retries are needed to eventually be successful,
+// This allows all our client calls to have this "retry" feature without explicitly wrapping them
+func (r *RetryRoundTripper) RoundTrip(req *http.Request) (resp *http.Response, err error) {
+	if errRetryOnError := retry.OnError(r.backoff, IsRetriable, func() error {
+		resp, err = r.transport.RoundTrip(req)
+		if err != nil && IsRetriable(err) {
+			r.log.Info("WARNING: retrying", "req.url", req.URL.String(), "error", err)
+		}
+		return err //nolint:wrapcheck
+	}); errRetryOnError != nil {
+		r.log.Info("WARNING: retrying failed", "error", errRetryOnError)
+	}
+
+	return resp, err //nolint:wrapcheck
```

Comment:
Should this be 'return resp, errRetryOnError'?

Reply:
Yeah you're right errRetryOnError and err should be the exact same value (at least with the impl they have now)...but I have it like this intentionally in this case. 

I just didn't want to pass on the error from `retry.OnError` instead just logging it...just so that from caller's perspective as if nothing changed (with or without retry). I'm open to changing though no worries. Wdyt?

Reply:
Yeah, right, they should be the same value. Just at first glance, my initial reation is to return errRetryOnError since that's the error from the retry function.